src/js/analyses/components/Create/ReferenceSelector.js

```diff
@@ -1,58 +1,94 @@
 import styled from "styled-components";
-import { Label, NoneFoundBox } from "../../../base";
+import { Label, NoneFoundBox, Input, Icon } from "../../../base";
 import { MultiSelector, MultiSelectorItem } from "../../../base/MultiSelector";
 import { Link } from "react-router-dom";
+import { useFuse } from "../../../base/hooks";
+import { SampleInputContainer, IconContainer, WorkflowItem } from "../../../subtraction/components/Selector";
 
 const StyledReferenceSelectorItem = styled(MultiSelectorItem)`
-    > span:last-child {
+    span:last-child {
         margin-left: auto;
     }
 `;
 
+const NameOverflowItem = styled.strong`
+    max-width: 45%;
+    overflow: hidden;
+    white-space: nowrap;
+`;
+
 export const ReferenceSelectorItem = ({ id, name, version }) => (
-    <StyledReferenceSelectorItem value={id}>
-        <strong>{name}</strong>
+    <StyledReferenceSelectorItem value={id} requireCheckbox={false}>
+        <NameOverflowItem>{name}</NameOverflowItem>
         <span>
             Index Version <Label>{version}</Label>
         </span>
     </StyledReferenceSelectorItem>
 );
 
 export const ReferenceSelector = ({ hasError, indexes, selected, onChange }) => {
-    const referenceComponents = map(indexes, index => (
-        <ReferenceSelectorItem
-            key={index.reference.id}
-            id={index.reference.id}
-            name={index.reference.name}
-            version={index.version}
-        />
-    ));
+    const [results, term, setTerm] = useFuse(indexes, ["reference.name"], [1]);
+
+    const references = results.filter(item => {
+        const obj = item.id ? item : item.item;
+        return !selected.includes(obj.reference.id);
+    });
 
-    if (!referenceComponents.length) {
+    const referenceComponents = references.map(item => {
+        const result = item.id ? item : item.item;
         return (
-            <>
-                <label htmlFor="references">References</label>
-                <NoneFoundBox noun="references" id="references">
-                    <Link to="/refs">Create one</Link>.
-                </NoneFoundBox>
-            </>
+            <Fragment key={item.id}>
+                {!selected.includes(result.reference.id) && (
+                    <ReferenceSelectorItem
+                        key={result.reference.id}
+                        id={result.reference.id}
+                        name={result.reference.name}
+                        version={result.version}
+                    />
+                )}
+            </Fragment>
         );
-    }
+    });
 
     return (
         <>
             <label>References</label>
-            <MultiSelector
-                error={hasError && "Reference(s) must be selected"}
-                noun="references"
-                selected={selected}
-                onChange={onChange}
-            >
-                {referenceComponents}
-            </MultiSelector>
+            {indexes.length - selected.length > 0 && (
```

Comment:
resolved. used referenceComponents.length < 0 && (

otherwise references0 renders when all references selected

Reply:
Ah yeah rendering the zero is less than ideal. an alternative option is to directly cast `referenceComponents.length` directly to a boolean via `Boolean(referenceComponents.length)` rather than doing a comparison with zero.

Reply:
This actually produces a different result. In this case indexes.length - selected.length is the remaining unselected references regardless of the search term. The referenceComponents length can be 0 if the search term yields no results. And in this case an empty search will result in the filter bar being taken away when it is still needed

Reply:
A fair point, that is going to be a problem. Another solution would be to filter the the indexes against the selected indexes prior to passing them to `useFuse`, which would give a convenient way to check, but given the need for indexing when a Fuse is created I don't actually think it is a great idea.

Unfortunately I still think the logic about what items are rendered is non-obvious. One off the cuff idea that I think would work would be to:

  1. Check if indexes.length is falsy, if it is then return the NoneFound message and don't execute anything else.
  2. Check if all the references are selected. Since we know from (1) that indexes.length >== 1 we can skip that check and just test if whether selected.length and index.length are the same. If they are, then return the `All references selected` message.
  3. Finally since we know that indexes.length >== 1 and that selected.length !== index.length then we can assume that the searchBar + options should be rendered. (Since it should never be the case that selected.length > index.length)

I would imagine this to be best implemented using three different return statements, but you could play around with that. Assuming it works, the advantage I see here is making the conditions which trigger each option pretty obvious at a glance. 

I am definitely not stuck on the exact logic flow either, so feel free to mess around and try to find something better.